### PR TITLE
client: New rpc method `getUncleCountByBlockNumber`

### PR DESCRIPTION
**Pull request by acolytec3**
Add new RPC method `eth_getUncleCountByBlockNumber` and related tests

### CODE REVIEW

**Comment by acolytec3** [comment]
cc @holgerd77

**Comment by jochem-brouwer** [comment]
Nice PR :+1: 

I see you have a lint commit, we usually squash these out:

```
git rebase -i HEAD~2 
git push -f
```

The rebase will launch an interactive CLI rebaser where you can rebase the latest two commits (`HEAD~2`). In the description you can see how you can squash/fixup the lint commit into the other commit. Then force push here (it's usually a good idea on force pushes to create a backup of the work).

**Comment by acolytec3** [comment]
@jochem-brouwer I reviewed the geth implementation and looks like your interpretation of `eth_coinbase` is correct and I just fundamentally misunderstood it.  As such, I have removed it from the PR as it looks like there is no back-end implementation for this at present within the monorepo so seems like a pointless method to expose until that back-end functionality is built out.

File: packages/client/lib/rpc/modules/eth.ts [hunk 1]
```diff
@@ -404,4 +404,34 @@ export class Eth {
   protocolVersion(_params = []) {
     return `0x${this.ethVersion.toString(16)}`
   }
+
+  /**
+   * Returns the current coinbase (i.e. the address that miner rewards are sent to)
+   * @param params An empty array
+   */
+  async coinBase() {
+    const header = await this._chain.getLatestHeader()
```
[the remark this answers is not in the document]
Comment (acolytec3) [reply packages/client/lib/rpc/modules/eth.ts]: I've removed this function as not applicable given the above note.

**Comment by jochem-brouwer** [comment]
Yep good point, it doesn't make much sense to include `eth_coinbase` if we don't have a miner implemented (even a very simple CPU one).

**Review by holgerd77 — APPROVED** [review]
LGTM

**Pull request merged**